src/braket/experimental/autoqasm/instructions/pulse_control.py

```diff
@@ -0,0 +1,152 @@
+
+
+def _pulse_instruction(name: str, frame: Frame, *args) -> None:
+    """Define a pulse control instruction.
+
+    Args:
+        name (str): Name of the pulse control instruction.
+        frame (Frame): Frame for which the instruction is apply to.
+    """
+    program_conversion_context = program.get_program_conversion_context()
+    program_conversion_context._has_pulse_control = True
+
+    pulse_sequence = PulseSequence()
+    pulse_sequence._program = program_conversion_context.get_oqpy_program()
+    with oqpy.Cal(pulse_sequence._program):
+        getattr(pulse_sequence, name)(frame, *args)
+
+
+def set_frequency(frame: Frame, frequency: float) -> None:
+    """Adds an instruction to set the frequency of the frame to the specified `frequency` value.
+
+    Args:
+        frame (Frame): Frame for which the frequency needs to be set.
+        frequency (float): Frequency value to set for the specified frame.
+    """
+    _pulse_instruction("set_frequency", frame, frequency)
+
+
+def shift_frequency(frame: Frame, frequency: float) -> None:
+    """Adds an instruction to shift the frequency of the frame by the specified `frequency` value.
+
+    Args:
+        frame (Frame): Frame for which the frequency needs to be shifted.
+        frequency (float): Frequency value by which to shift the frequency for the specified frame.
+    """
+    _pulse_instruction("shift_frequency", frame, frequency)
+
+
+def set_phase(frame: Frame, phase: float) -> None:
+    """Adds an instruction to set the phase of the frame to the specified `phase` value.
+
+    Args:
+        frame (Frame): Frame for which the frequency needs to be set.
+        phase (float): Phase value to set for the specified frame.
+    """
+    _pulse_instruction("set_phase", frame, phase)
+
+
+def shift_phase(frame: Frame, phase: float) -> None:
+    """Adds an instruction to shift the phase of the frame by the specified `phase` value.
+
+    Args:
+        frame (Frame): Frame for which the phase needs to be shifted.
+        phase (float): Phase value by which to shift the phase for the specified frame.
+    """
+    _pulse_instruction("shift_phase", frame, phase)
+
+
+def set_scale(frame: Frame, scale: float) -> None:
+    """Adds an instruction to set the scale on the frame to the specified `scale` value.
+
+    Args:
+        frame (Frame): Frame for which the scale needs to be set.
+        scale (float): scale value to set on the specified frame.
+    """
+    _pulse_instruction("set_scale", frame, scale)
+
+
+def play(frame: Frame, waveform: Waveform) -> None:
+    """Adds an instruction to play the specified waveform on the supplied frame.
+
+    Args:
+        frame (Frame): Frame on which the specified waveform signal would be output.
+        waveform (Waveform): Waveform envelope specifying the signal to output on the specified
+            frame.
+    """
+    _pulse_instruction("play", frame, waveform)
+
+
+def capture_v0(frame: Frame) -> None:
```

Comment:
What does v0 mean here?

Reply:
It's using the openqasm definition: https://openqasm.com/language/openpulse.html#capture-instruction, and also aligning with what we do in BDK.

Reply:
We could rename it based on what its functionality, but haven't started this conversation

Reply:
Yeah, let's avoid using numbers that require specialized knowledge. For now, could we call it `capture`?

Reply:
It turns out `capture` is reserve for a feature in the future. We could rename `capture_v0` to something more descriptive, e.g., `capture_to_buffer`, but it may raise more questions than the benefit it gets: what is a buffer? how is it different from BDK's and openqasm's `capture_v0` and why we rename it? The ultimate definition is in openqasm spec. Having name mappings between different modules would cause confusion